d8s_python/ast.py

```diff
@@ -0,0 +1,205 @@
+                return self.exception.exc.func.id
+            elif hasattr(
+                self.exception.exc.func, 'attr'
+            ):  # this handles ast.Raise nodes raising a non-built-in error (e.g. "pint.UndefinedUnitError")
+                return f'{self.exception.exc.func.value.id}.{self.exception.exc.func.attr}'
+        elif hasattr(self.exception, 'type') and self.exception.type:  # this handles ast.ExceptHandler nodes
+            if hasattr(
+                self.exception.type, 'id'
+            ):  # this handles ast.ExceptHandler nodes raising a built-in error (e.g. "RuntimeError")
+                return self.exception.type.id
+            elif hasattr(
+                self.exception.type, 'attr'
+            ):  # this handles ast.ExceptHandler nodes raising a non-built-in error (e.g. "pint.UndefinedUnitError")
+                return f'{self.exception.type.value.id}.{self.exception.type.attr}'
+        elif hasattr(
+            self.exception, 'attr'
+        ):  # this handles situations where the exception being raised is an ast.Attribute (e.g. "pint.UndefinedUnitError")
+            return f'{self.exception.value.id}.{self.exception.attr}'
+        elif hasattr(self.exception, 'id'):  # this handles situations where the exception being raised is an ast.Name (e.g. "e")
+            return self.exception.id
+
+    @property
+    def exceptions_handled_by_handler(self) -> Optional[Iterable[str]]:
+        """Return all of the exceptions handled by the given exception handler."""
+
+        if self.exception_handler.type and hasattr(self.exception_handler.type, 'elts'):
+            yield from (
+                python_ast_exceptions(i, None).name
+                for i in self.exception_handler.type.elts
+            )
+
+        elif exception_name := python_ast_exceptions(self.exception_handler, None).name: yield exception_name
+
+    @property
+    def exceptions_raised_by_handler(self) -> Optional[Iterable[str]]:
+        """Return the exception raised by the given exception handler."""
+
+        from d8s_lists import iterable_replace
+
+        exceptions_names = list(map(python_ast_raise_name, python_ast(self.exception_handler).get_astObjects_by_type(ast.Raise)))
+
+        for name in exceptions_names:
+            if name == handler.name:
+                exceptions_names = iterable_replace(
+                    exceptions_names,
+                    name,
+                    self.exceptions_handled
+                )
+            elif name is None:
+                exceptions_names = iterable_replace(
+                    exceptions_names,
+                    name,
+                    self.exceptions_handled
+                )
+
+        import more_itertools
+
+        yield from more_itertools.collapse(
+            exceptions_names,
+            base_type=str
+        )
+
+
+class python_ast:
+    def __init__(self, astree: object): self.astree = astree
```

Comment:
Sorry, but I don't understand the value of wrapping these functions in classes. What's the benefit of this?

For example, if someone (let's calling her "Alice") wants to use the `python_ast_objects_of_type` she can currently do:

```python
python_ast_objects_of_type(code, ast_type)
```

With this PR, she would have to do:

```python
python_ast(code).get_astObjects_by_type(ast_type)
```

How this the second one better?

I see that declaring a class lets you declare the class once and then run multiple functions like:

```python
a = python_ast(code)
a.get_astObjects_by_type(ast_type)
a.get_lines()
```

but similar functionality can be achieved using the current library:

```
as_tree = python_ast_parse(code)
python_ast_objects_of_type(as_tree)
python_ast_object_line_numbers(as_tree)
```

I'm open to this PR, but want to better understand it's benefits. I think we can certainly simplifying the names of the functions to reduce their verbosity.